### PR TITLE
Refactored varible names for promoted content endpoint

### PR DESCRIPTION
**Pull request by davidthomas22**

### CODE REVIEW

**Review by spikeheap — COMMENTED** [review]
Ooh, I know this is a draft at the moment, but ❤️ for small PRs. This one's nice and easy to see the whole effect of the change 👍. Also loving the meaningful variable names!

File: modules/custom/moj_resources/src/PromotedContentApiClass.php [hunk 1]
```diff
@@ -153,20 +153,23 @@ private function promotedTags($prison)
...
-    $terms = $this->term_storage->loadMultiple($termIds);
+    $loadedTerms = $this->term_storage->loadMultiple($termIds);
 
-    $promotedTerms = array_filter($terms, function ($item) use ($prison) {
-      if ($item->field_moj_promoted->value == true && $prison == $item->field_promoted_to_prison->target_id) {
+    $promotedTerms = array_filter($loadedTerms, function ($term) use ($prison) {
+      $isPromotedTerm = $term->field_moj_promoted->value == true;
+      $promotedPrisonForTerm = $term->field_promoted_to_prison->target_id;
+      $termIsPromotedToThisPrison = $prison == $promotedPrison;
+      if ($isPromotedTerm && $termIsPromotedToThisPrison) {
```
Comment (psoleckimoj) [inline modules/custom/moj_resources/src/PromotedContentApiClass.php:170]: These can be combined as the logic is all AND, so

`if ($isPromotedTerm && ($termIsPromotedToThisPrison || !$promotedPrisonForTerm)) {`

Comment (psoleckimoj) [reply modules/custom/moj_resources/src/PromotedContentApiClass.php:170]: I just hate seeing two `return true;` straight after one another 😁

**Review by LucasCairns — APPROVED** [review]
This all looks good to me, good to see dead code removed 👍

**Review by psoleckimoj — APPROVED** [review]
LGTM, kill it, kill it now!

**Pull request closed without merging**